tests/components/zeversolar/test_init.py

```diff
@@ -1,32 +1,96 @@
 """Test the init file code."""
 
+from unittest.mock import patch
+
 import pytest
+from zeversolar import StatusEnum, ZeverSolarData
+from zeversolar.exceptions import ZeverSolarTimeout
 
 import homeassistant.components.zeversolar.__init__ as init
 from homeassistant.components.zeversolar.const import DOMAIN
-from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
 from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import ConfigEntryNotReady
 
-from tests.common import MockConfigEntry, MockModule, mock_integration
+from tests.common import MockConfigEntry
 
 MOCK_HOST_ZEVERSOLAR = "zeversolar-fake-host"
 MOCK_PORT_ZEVERSOLAR = 10200
 
+MOCK_DATA = ZeverSolarData(
+    wifi_enabled=False,
+    serial_or_registry_id="1223",
+    registry_key="A-2",
+    hardware_version="M10",
+    software_version="123-23",
+    reported_datetime="19900101 23:00",
+    communication_status=StatusEnum.OK,
+    num_inverters=1,
+    serial_number="123456778",
+    pac=1234,
+    energy_today=123,
+    status=StatusEnum.OK,
+    meter_status=StatusEnum.OK,
+)
 
-async def test_async_setup_entry_fails(hass: HomeAssistant) -> None:
-    """Test the sensor setup."""
-    mock_integration(hass, MockModule(DOMAIN))
 
-    config = MockConfigEntry(
+def create_config_mock():
```

Comment:
Create a fixture (maybe move it to `conftest.py`)
```suggestion
@pytest.fixture
def config_entry() -> MockConfigEntry:
```